### PR TITLE
fix(message validation): validate duplicated topics

### PR DESCRIPTION
**Pull request by thalesmg**
Fixes https://emqx.atlassian.net/browse/EMQX-12254

Release version: e5.7

## PR Checklist
Please convert it to a draft if any of the following conditions are not met. Reviewers may skip over until all the items are checked:

- [x] Added tests for the changes
- [ ] Added property-based tests for code which performs user input validation
- [ ] Changed lines covered in coverage report
- [x] Change log has been added to `changes/(ce|ee)/(feat|perf|fix|breaking)-<PR-id>.en.md` files
- [x] For internal contributor: there is a jira ticket to track this change
- [ ] Created PR to [emqx-docs](https://github.com/emqx/emqx-docs) if documentation update is required, or link to a follow-up jira ticket
- [ ] Schema changes are backward compatible

## Checklist for CI (.github/workflows) changes

- [ ] If changed package build workflow, pass [this action](https://github.com/emqx/emqx/actions/workflows/build_packages.yaml) (manual trigger)
- [ ] Change log has been added to `changes/` dir for user-facing artifacts update

### CODE REVIEW

File: apps/emqx_message_validation/src/emqx_message_validation_schema.erl [hunk 1]
```diff
@@ -269,3 +270,23 @@ do_validate_unique_schema_checks(
     end;
 do_validate_unique_schema_checks([_Check | Rest], Seen, Duplicated) ->
     do_validate_unique_schema_checks(Rest, Seen, Duplicated).
+
+validate_unique_topics(Topics) ->
+    Grouped = maps:groups_from_list(
```
Comment (zmstone) [inline apps/emqx_message_validation/src/emqx_message_validation_schema.erl:275]: we'll need to update README to drop OTP 24 for this change ?

File: apps/emqx_message_validation/src/emqx_message_validation_schema.erl [hunk 2]
```diff
@@ -269,3 +270,23 @@ do_validate_unique_schema_checks(
...
+        fun(_T, Ts) -> length(Ts) > 1 end,
+        Grouped
+    ),
+    case maps:keys(DuplicatedMap) of
+        [] ->
+            ok;
+        Duplicated ->
+            Msg = iolist_to_binary([
+                <<"duplicated topics: ">>,
+                lists:join(" ; ", Duplicated)
```
Comment (zmstone) [inline apps/emqx_message_validation/src/emqx_message_validation_schema.erl:289]: nit:
```suggestion
                lists:join(", ", Duplicated)
```

File: apps/emqx_message_validation/src/emqx_message_validation_schema.erl (hunk 1 above)
Comment (thalesmg) [reply apps/emqx_message_validation/src/emqx_message_validation_schema.erl:275]: I can update it, but this is not new: several other places already use `groups_from_list` :thinking: 

e.g.: https://github.com/emqx/emqx/blob/002dc8541ba9aad78aea48b4cb336c6b1de41cf7/apps/emqx_management/src/emqx_mgmt_api_trace.erl#L567

**Review by zmstone — APPROVED** [review]

**Comment by yanzhiemq** [comment]
[TestExecution](https://emqx.atlassian.net/browse/QA-2154)

**Pull request merged**